api/src/main/java/javax/mvc/locale/LocaleResolverContext.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * http://glassfish.java.net/public/CDDL+GPL_1_1.html
+ * or packager/legal/LICENSE.txt.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at packager/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * Oracle designates this particular file as subject to the "Classpath"
+ * exception as provided by Oracle in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package javax.mvc.locale;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Configuration;
+
+/**
+ * <p>Contextual data used by a {@link javax.mvc.locale.LocaleResolver} to resolve the request
+ * locale. This includes the request from the Servlet container, the JAX-RS configuration,
+ * among other data.</p>
+ *
+ * @author Christian Kaltepoth
+ * @see javax.mvc.locale.LocaleResolver
+ * @see javax.servlet.http.HttpServletRequest
+ * @since 1.0
+ */
+public interface LocaleResolverContext {
+
+    /**
+     * Returns HTTP request object from the Servlet container.
+     *
+     * @return HTTP request object.
+     */
+    HttpServletRequest getRequest();
```

Comment:
Exactly. That's my point. So perhaps we should remove `HttpServletRequest` and add all the other stuff we get in case of a pre-matching filter from `ContainerRequestContext` (especially `UriInfo`, `Request`, etc)? If implementations need `HttpServletRequest`, they can inject it and we don't have the `HttpServletRequest` dependency in the API.